### ISSUE TITLE
Support loading template from aws s3 private bucket

### ISSUE THREAD

**Issue by Zheaoli** [lima-vm/lima#1087]
### Description

We sometimes need to store our template in AWS s3 private bucket. So maybe we can support `limactl start --name=default template+s3://${s3Url}` and read the credit from the AWS config directory

**Comment by jandubois** [comment]
I don't like the idea of creating a dependency on the AWS SDK in Lima.

You can also already do what you want with this command (from bash):
```
limactl start --name=default <(aws s3 cp s3://${s3Url} -)
```

**Comment by adamcstephens** [comment]
I think more interesting than the template itself would be support for reading the images from s3. Which doesn't appear to be possible with just shell redirects.

**Comment by jandubois** [comment]
> I think more interesting than the template itself would be support for reading the images from s3. Which doesn't appear to be possible with just shell redirects.

I think you just have to use the https: URL for the object. I don't know if accessing an object that doesn't have a URL is a reasonable use case; why not provide a URL for the image?

**Comment by adamcstephens** [comment]
Accessing an object with unauthenticated https requires making it available anonymously, no? Thus it’s no longer a private bucket.

**Comment by jandubois** [comment]
> Thus it’s no longer a private bucket.

Yes. I think you can generate pre-signed URLs for private objects, but they will expire. Also means you cannot share the URLs publicly.

My objection to adding a dependency on the AWS SDK still stands. I think the way to do this would be:

1. Extend the custom URL scheme to be able to return the content, and not just a replacement URL
2. Implement support for custom URL schemes for images as well
